samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/shoppingcarts/config/CommandsConfig.java

```diff
@@ -0,0 +1,98 @@
+import io.eventdriven.ecommerce.pricing.ProductPriceCalculator;
+import io.eventdriven.ecommerce.pricing.RandomProductPriceCalculator;
+import io.eventdriven.ecommerce.shoppingcarts.Events;
+import io.eventdriven.ecommerce.shoppingcarts.ShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.addingproductitem.AddProductItemToShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.canceling.CancelShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.confirming.ConfirmShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.opening.OpenShoppingCart;
+import io.eventdriven.ecommerce.shoppingcarts.removingproductitem.RemoveProductItemFromShoppingCart;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.context.annotation.ApplicationScope;
+import org.springframework.web.context.annotation.RequestScope;
+
+import java.util.Optional;
+
+@Configuration
+public class CommandsConfig {
+  @Bean
+  @RequestScope
+  public CommandHandler<OpenShoppingCart> handleInitializeShoppingCart(EntityStore<ShoppingCart> store) {
+    return command ->
+      store.Add(
+        () -> OpenShoppingCart.Handle(command),
+        command.shoppingCartId()
+      );
+  }
+
+  @Bean
+  @RequestScope
+  CommandHandler<AddProductItemToShoppingCart> handleAddProductItemToShoppingCart(
+    EntityStore<ShoppingCart> store,
+    ProductPriceCalculator productPriceCalculator
+  ) {
+    return command ->
+      store.GetAndUpdate(
+        current -> AddProductItemToShoppingCart.Handle(productPriceCalculator, command, current),
+        command.shoppingCartId(),
+        Optional.of(command.expectedVersion())
+      );
+  }
+
+
+  @Bean
+  @RequestScope
+  CommandHandler<RemoveProductItemFromShoppingCart> handleRemoveProductItemFromShoppingCart(EntityStore<ShoppingCart> store) {
+    return command ->
+      store.GetAndUpdate(
+        current -> RemoveProductItemFromShoppingCart.Handle(command, current),
+        command.shoppingCartId(),
+        Optional.of(command.expectedVersion())
+      );
+  }
+
+  @Bean
+  @RequestScope
+  CommandHandler<ConfirmShoppingCart> handleConfirmShoppingCart(EntityStore<ShoppingCart> store) {
+    return command ->
+      store.GetAndUpdate(
+        current -> ConfirmShoppingCart.Handle(command, current),
+        command.shoppingCartId(),
+        Optional.of(command.expectedVersion())
+      );
+  }
+
+  @Bean
+  @RequestScope
+  CommandHandler<CancelShoppingCart> handleCancelShoppingCart(EntityStore<ShoppingCart> store) {
+    return command ->
+      store.GetAndUpdate(
+        current -> CancelShoppingCart.Handle(command, current),
+        command.shoppingCartId(),
+        Optional.of(command.expectedVersion())
+      );
+  }
+
+  @Bean
+  @ApplicationScope
+  ProductPriceCalculator productPriceCalculator() {
+    return new RandomProductPriceCalculator();
+  }
+
+  @Bean
+  @ApplicationScope
```

Comment:
a to nie może być zwykły bean?

Reply:
Być może to moja naleciałość z .NETa, ale staram się rejestrować zawsze w takim scopie w jakim oczekuję, że będzie jego cykl życia. Akurat ta dummy implementacja spokojnie może być Singletonem, stąd też i `ApplicationScope`. Czy w Javie się to robi inaczej?

Reply:
Btw. "zmerdżowałem" te klasy w application service, zgodnie z Twoją sugestią w https://github.com/oskardudycz/EventSourcing.JVM/pull/1/commits/dbc9ca7fc2fea6f78bebfc107eb28fc88fac7ed1. Faktycznie taki podział w tym kontekście to był przerost formy nad treścią i zaciemniał obraz.